### PR TITLE
Add ext- requirements to composer.json

### PR DESCRIPTION
**Pull request by misakstvanu**
Having extensions listed in documentation is cool, having them in composer is even cooler. Added bcmath, gmp, json to composer.json.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -11,6 +11,9 @@
     ],
     "require": {
         "php": ">=7.4",
+        "ext-bcmath": "*",
+        "ext-gmp": "*",
+        "ext-json": "*",
```
Comment (matthewelwell) [inline composer.json:16]: Does JSON need adding here since it wasn't in the original documentation?

Comment (matthewelwell) [inline composer.json:16]: Why `"*"` all of these? Shouldn't we define the versions?

Comment (misakstvanu) [reply composer.json:16]: External dependencies in composer are commonly used just as an "check whether the tool is installed and enabled" if you don't need any specific or new feature.

Comment (matthewelwell) [reply composer.json:16]: Ah, ok, yeah. In fact, I just removed PHP7 from the test matrix [here](https://github.com/Flagsmith/flagsmith-php-client/pull/80), so I think we can probably remove JSON from this list.

Comment (misakstvanu) [reply composer.json:16]: Fair, sorry about that.

**Review by matthewelwell — APPROVED** [review]
@misakstvanu can you fix the conflicts and I can get this merged?

**Comment by misakstvanu** [comment]
Done

**Comment by VincentLanglet** [comment]
Hi @matthewelwell,

There is something weird with the "4.5.0 release" with this PR.
The tag is available here: https://github.com/Flagsmith/flagsmith-php-client/releases/tag/v4.5.0
But not on packagist https://packagist.org/packages/flagsmith/flagsmith-php-client

So we cannot install the `4.5.0` version of flagsmith.

**Comment by matthewelwell** [comment]
Sorry @VincentLanglet , I believe that's because we hadn't updated the version number in composer.json 🤦 

I have a PR open to do that now ([here](https://github.com/Flagsmith/flagsmith-php-client/pull/87)), and I will recreate the tag as soon as I can. Apologies.

**Pull request merged**